crates/utils/re_arrow_util/src/lib.rs

```diff
@@ -384,3 +384,87 @@ where
     array.shrink_to_fit(); // VERY IMPORTANT! https://github.com/rerun-io/rerun/issues/7222
     array
 }
+
+// ----------------------------------------------------------------------------
+
+/// Convert `[A, B, null, D, …]` into `[[A], [B], null, [D], …]`
+pub fn wrap_in_list_array(field: &Field, array: ArrayRef) -> (Field, ListArray) {
+    re_tracing::profile_function!();
+
+    debug_assert_eq!(field.data_type(), array.data_type());
+
+    let item_field = Arc::new(Field::new(
+        "item",
+        field.data_type().clone(),
+        field.is_nullable(), // TODO(emilk): it would be nice to remove the "inner nullability", and just have outer nullability.
```

Comment:
I'm not sure how to strip the input `array` from its null buffer, and I don't think it matters much (there is no memory overhead)

Reply:
That makes sense - I'll update the comment!